components/server/src/authorization/authorizer.ts

```diff
@@ -503,7 +525,61 @@ export class Authorizer {
     }
 }
 
-export async function isFgaChecksEnabled(userId: string | undefined): Promise<boolean> {
+async function getSubjectFromCtx(passed: Subject): Promise<SubjectId> {
+    const ctxSubjectId = ctxTrySubjectId();
+    const ctxUserId = ctxSubjectId?.userId();
+
+    const passedSubjectId = Subject.toId(passed);
+    const passedUserId = passedSubjectId.userId();
+
+    // Check: Do the subjectIds match?
+    const matchingSubjectId = ctxUserId === passedUserId;
+    const match = !ctxUserId ? "ctx-user-id-missing" : matchingSubjectId ? "match" : "mismatch";
+    reportAuthorizerSubjectId(match);
+    if (match !== "match") {
+        try {
+            // Get hold of the stack trace
+            throw new Error("Authorizer: SubjectId mismatch");
+        } catch (err) {
+            log.error("Authorizer: SubjectId mismatch", err, {
+                match,
+                ctxUserId,
+                passedUserId,
+            });
+        }
+    }
+
+    // Check feature flag, based on the passed subjectId
+    const authViaContext = await getExperimentsClientForBackend().getValueAsync("authWithRequestContext", false, {
+        user: ctxUserId
+            ? {
+                  id: ctxUserId,
+              }
+            : undefined,
+    });
+    if (!authViaContext) {
+        return passedSubjectId;
+    }
+
+    if (!ctxSubjectId || match === "mismatch") {
+        const err = new ApplicationError(ErrorCodes.NOT_AUTHENTICATED, `Cannot authorize request`);
```

Comment:
Should it be PERMISSION_DENIED insted? Not sure, it seems we have a subject, i.e. it is authenticated

Reply:
I was a bit torn (the two conditions are different things if you look closer). But it better fits the contract of the Authorizer to return PERMISSION_DENIED, so going with that. :+1: